src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -586,7 +586,20 @@ public override BoundNode VisitConversion(BoundConversion node)
             switch (node.ConversionKind)
             {
                 case ConversionKind.MethodGroup:
-                    VisitMethodGroup((BoundMethodGroup)node.Operand, parentIsConversion: true);
+                    var methodGroup = (BoundMethodGroup)node.Operand;
+                    VisitMethodGroup(methodGroup, parentIsConversion: true);
+
+                    MethodSymbol method = node.Conversion.Method;
```

Comment:
>MethodSymbol method = node.Conversion.Method; [](start = 20, length = 45)

Could you please explain in details what triggers the undesired diagnostic? Perhaps, things in the bound tree that trigger it, shouldn't be there in the first place and the right fix would be to adjust the tree produced by initial binding rather than trying to special case the situation here.

Reply:
@AlekseyTs Consider the situation where you have two overloads of `M`, one instance and one static, and you use method group conversion to access the static one. In that situation, the bound tree looks like this:

```
BoundConversion
 ├─Conversion.Method: static M
 └─Operand: BoundMethodGroup
    ├─Methods: [ instance M, static M ]
    └─ReceiverOpt: BoundThisReference
```

The problem, as I understand it, is that `BoundMethodGroup` does not know which overload will be chosen, so it always has a synthesized `this` receiver in these cases.

I don't see a simple way to adjust the bound tree in a way that would avoid this problem. E.g. having no receiver if no method requires a receiver would not really help, because the case where you're overloading static and instance methods also has to be handled somehow. Some kind of larger change to the tree could probably help, but I don't think that's warranted.